spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/ZuulProxyAutoConfiguration.java

```diff
@@ -86,10 +83,9 @@ public HasFeatures zuulFeature() {
 
 	@Bean
 	@ConditionalOnMissingBean(DiscoveryClientRouteLocator.class)
-	public DiscoveryClientRouteLocator discoveryRouteLocator() {
+	public DiscoveryClientRouteLocator discoveryRouteLocator(ServiceRouteMapper serviceRouteMapper) {
 		return new DiscoveryClientRouteLocator(this.server.getServlet().getContextPath(),
-				this.discovery, this.zuulProperties, this.serviceRouteMapper,
```

Comment:
Why did you remove it from the constructor?

Reply:
You're a fast one. ;) It's still a draft PR.